plugins/modules/host_template.py

```diff
@@ -0,0 +1,209 @@
+
+ANSIBLE_METADATA = {
+    "metadata_version": "1.1",
+    "status": ["preview"],
+    "supported_by": "community",
+}
+
+DOCUMENTATION = r"""
+---
+module: host_template
+short_description: Configure a host template
+description:
+  - Creates a new host template or updates an existing one
+  - The module supports C(check_mode).
+author:
+  - "Ronald Suplina (@rsuplina)"
+requirements:
+  - cm_client
+options:
+  name:
+    description:
+      - The associated cluster name.
+    type: str
+    required: yes
+    aliases:
+      - cluster_name
+  host_template_name:
+    description:
+      - The name of the host template.
+    type: str
+    required: yes
+  roleConfigGroupRefs:
+    description:
+      - The names of the role config groups
+    type: list
+    returned: yes
+"""
+
+EXAMPLES = r"""
+---
+- name: Create host template
+  cloudera.cluster.host_template
+    host: example.cloudera.com
+    username: "jane_smith"
+    password: "S&peR4Ec*re"
+    name: "base_cluster"
+    role_configs_groups: ["kafka-GATEWAY-BASE", "atlas-ATLAS_SERVER-BASE" , "hive_on_tez-GATEWAY-BASE"]
+
+- name: Update host template
+  cloudera.cluster.host_template
+    host: example.cloudera.com
+    username: "jane_smith"
+    password: "S&peR4Ec*re"
+    name: "base_cluster"
+    role_configs_groups: ["kafka-GATEWAY-BASE", "atlas-ATLAS_SERVER-BASE"]
+"""
+
+RETURN = r"""
+---
+cloudera_manager:
+  description:
+    - Retrieve details about host template.
+  type: dict
+  elements: dict
+  returned: always
+  contains:
+    name:
+      description:
+        - The name of the host template
+      type: str
+      returned: always
+    clusterRef:
+      description: A reference to the enclosing cluster.
+      type: dict
+      returned: always
+    roleConfigGroupRefs:
+      description:
+        - The role config groups belonging to this host tempalte.
+      type: list
+      returned: always
+"""
+
+
+class ClouderaHostTemplate(ClouderaManagerModule):
+    def __init__(self, module):
+        super(ClouderaHostTemplate, self).__init__(module)
+
+        # Set the parameters
+        self.cluster_name = self.get_param("name")
+        self.host_template_name = self.get_param("host_template_name")
+        self.role_configs_groups = self.get_param("role_configs_groups")
+
+        # Initialize the return value
+        self.host_template = []
+        self.host_template_output = []
+
+        # Execute the logic
+        self.process()
+
+    @ClouderaManagerModule.handle_process
+    def process(self):
+        host_temp_api_instance = HostTemplatesResourceApi(self.api_client)
+
+        try:
+            ClustersResourceApi(self.api_client).read_cluster(self.cluster_name)
+        except ApiException as ex:
+            if ex.status == 404:
+                self.module.fail_json(
+                    msg="Cluster does not exist: " + self.cluster_name
+                )
+            else:
+                raise ex
+
+        if not self.module.check_mode:
```

Comment:
This condition is too early in the module logic.  We want to capture any potential changes to the host template and report them via the `diff` object.